inc/class_parser.php

```diff
@@ -1605,18 +1605,27 @@ function mycode_parse_video_disabled_callback($matches)
 	*/
 	function mycode_auto_url($message)
 	{
-		$message = " ".$message;
-
-		if($this->options['allow_url'] != 1)
-		{
-			return $message;
-		}
-		
 		// Links should end with slashes, numbers, characters and braces but not with dots, commas or question marks
 		// Don't create links within existing links (handled up-front in the callback function).
-		$message = preg_replace_callback("#<a\\s[^>]*>.*?</a>|([\s\(\)\[\>])(http|https|ftp|news|irc|ircs|irc6){1}(://)([^\/\"\s\<\[\.]+\.([^\/\"\s\<\[\.]+\.)*[\w]+(:[0-9]+)?(/([^\"\s<\[]|\[\])*)?([\w\/\)]))(?![^<>]*?>)#ius", array($this, 'mycode_auto_url_callback'), $message);
-		$message = preg_replace_callback("#<a\\s[^>]*>.*?</a>|([\s\(\)\[\>])(www|ftp)(\.)(([^\/\"\s\<\[\.]+\.)*[\w]+(:[0-9]+)?(/([^\"\s<\[]|\[\])*)?([\w\/\)]))(?![^<>]*?>)#ius", array($this, 'mycode_auto_url_callback'), $message);
-		$message = my_substr($message, 1);
+		$message = preg_replace_callback(
+			"~
+				<a\\s[^>]*>.*?</a>|								# match and return existing links
+				(?<=^|[\s\(\)\[\>])								# character preceding the link
+				(?P<prefix>
+					(?:http|https|ftp|news|irc|ircs|irc6)://|	# scheme, or
```

Comment:
I take it these comments are supposed to be lined up? if so, they aren't any longer based on the GitHub review UI. If they're not supposed to be lined up, can we collapse the whitespace a bit?

Reply:
Yes, lined up - try appending `?ts=4` to GitHub's file preview URL.

Setting up a `.editorconfig` file like on the 1.9 branch should make the GitHub UI display these correctly: https://github.com/mybb/mybb/blob/6d6c8bfdd56fa8df2d1cd4ef68696805ab690850/.editorconfig#L17-L19